subgraphs/balancer-v2/tests/handlers.test.ts

```diff
@@ -0,0 +1,229 @@
+  assert.equals(
+    ethereum.Value.fromSignedBigIntArray(pool.inputTokenBalances),
+    ethereum.Value.fromSignedBigIntArray(newAmounts),
+  );
+});
+
+/**
+ * Amounts inspired from https://etherscan.io/tx/0x792c49770ded77cb7bd2b1e9b2348431e2ec4b34fd93bbd0a17f854277566bfc
+ */
+test("Handle swap and updates base asset usd price value", () => {
+  let amountIn = BigInt.fromString("6807327166843002652");
+  let amountOut = BigInt.fromString("20490320269");
+
+  let pool = LiquidityPool.load(usdcWethPoolId.toHexString());
+  if (pool == null) throw new Error("Pool is not defined");
+  let newAmounts = [pool.inputTokenBalances[0].minus(amountOut), pool.inputTokenBalances[1].plus(amountIn)];
+
+  const swap = createNewSwapEvent(
+    usdcWethPoolId,
+    Address.fromString(weth.id),
+    Address.fromString(usdc.id),
+    amountIn,
+    amountOut,
+  );
+  handleSwap(swap);
+  pool = LiquidityPool.load(usdcWethPoolId.toHexString());
+  if (pool == null) throw new Error("Pool is not defined");
+  assert.equals(
+    ethereum.Value.fromSignedBigIntArray(pool.inputTokenBalances),
+    ethereum.Value.fromSignedBigIntArray(newAmounts),
+  );
+
+  let token = _TokenPrice.load(weth.id);
+  if (!token) throw Error("Token price is not defined");
+
+  assert.stringEquals(token.lastUsdPrice.toString(), expectedWethPrice);
+
+  pool = LiquidityPool.load(usdcWethPoolId.toHexString());
+  if (pool == null) throw new Error("Pool is not defined");
+  assert.stringEquals(pool.totalValueLockedUSD.toString(), expectedTVL);
+  log.info(pool.totalVolumeUSD.toString(), [])
+});
+
+test("Pool with stable and not base asset", () => {
+  let registerPoolEvent = createNewPoolEvent(umaUsdcPoolId, umaUsdcPoolAddress, 2);
+
+  const pair = [Address.fromString(uma.id), Address.fromString(usdc.id)];
+
+  const tokensRegisterEvent = createTokensRegisteredEvent(umaUsdcPoolId, pair, []);
+
+  handlePoolRegister(registerPoolEvent);
+  handleTokensRegister(tokensRegisterEvent);
+
+  const depositAmounts = [BigInt.fromString("123804125414775870192714"), BigInt.fromString("786018294699")];
+
+  const deposit = createNewPoolBalanceChangeEvent(
+    umaUsdcPoolId,
+    Address.fromString("0xf71d161fdc3895f21612d79f15aa819b7a3d296a"),
+    [Address.fromString(uma.id), Address.fromString(usdc.id)],
+    depositAmounts,
+    [new BigInt(0), new BigInt(0)],
+  );
+
+  handlePoolBalanceChanged(deposit);
+  let amountIn = BigInt.fromString("1107936999999999800000");
+  let amountOut = BigInt.fromString("7069293085");
+
+  const swap = createNewSwapEvent(
+    umaUsdcPoolId,
+    Address.fromString(uma.id),
+    Address.fromString(usdc.id),
+    amountIn,
+    amountOut,
+  );
+  handleSwap(swap);
+  let token = _TokenPrice.load(uma.id.toLowerCase());
+  if (token == null) throw Error("Token price is not defined");
+
+  assert.stringEquals(token.lastUsdPrice.toString(), expectedUmaPrice);
+});
+
+// test("Pool with weth and weight with low liquidity: Weth is the out token", () => {
```

Comment:
why is this commented out?

Reply:
This has been commented out because now it's interacting with the logic of the oracle written by Phoenix. I will also mock the contract of yearn lens to expect the correct behavior